presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -823,4 +823,27 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
      * <p>
      */
     default void validateScan(ConnectorSession session, ConnectorTableHandle handle) {}
+
+    /**
+     * API to determine if a table state (data in the table) has changed compared to cached table state in Presto.
+     * The cached table in Presto may take the form of result cache or a materialized view.
+     */
+    /**
+     * Method to request the connector for a token that captures the state of the table given the filters.
+     * Engine does not interpret the token in any way. It simply presents it back to the connector to determine freshness of the cached table
+     * the token is associated with.
+     */
+    default Optional<TableToken> getTableToken(ConnectorSession session, ConnectorTableHandle tableHandle, TupleDomain<ColumnHandle> predicate)
```

Comment:
The predicate should be unnecessary. The table handle should encompass all the pushed down predicates, projections or other more complex operations via the `applyXXX` calls.

Reply:
You are right. The 'predicate' parameter is not necessary.